step/src/main/webapp/chat.jsp

```diff
@@ -17,53 +17,113 @@ limitations under the License.
-          <form action="Chat" class="form-container" method="POST">
-            <h1> Chat </h1>
-            <label for="text"><b>Message</b></label>
-            <br>
-            <c:forEach items="${messages}" var="m">
-                <tr>
-                    <td><c:out value="${m.getSenderID()}"/> to </td>
-                    <td><c:out value="${m.getRecipientID()}"/>: </td>
-                    <td><c:out value="${m.getText()}"/></td>
-                    <td><c:out value="${m.timestamp()}"/></td>
-                </tr>
-            </c:forEach>
-            <% String user = request.getParameter("user"); %>
-            <input type="hidden" name="user" value="<%= user %>"> </input>
-            <textarea placeholder="Type message.." name="text" required></textarea>
-
-            <button type="submit" class="btn">Send</button>
-            <button type="button" class="btn cancel" onclick="closeFrom()">Close</button>
-          </form> 
-      </div>
-
-      <script>
-            function openForm() {
-                document.getElementById("myForm").style.display = "block";
+    <head> 
+        <meta charset="UTF-8">
+        <title> Friend Matching Plus </title>
+        <link rel="stylesheet" href="style_chat1.css"> 
+        <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.0/css/bootstrap.min.css">
+        <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css">
+    </head>
+    <body>
+        <div id="chat-container">
+            <div id="search-container">
+                <input type="text" placeholder="Search" />
+            </div>
+            <div id="conversation-list" class="tab"> 
+                <c:forEach items="${matches}" var="match">
+                    <tr>
+                        <td><div class="message-text"></div></td>
+                        <td><div class="message-time"></div></td>
+                    </tr>
+                    <form action="Chat" method="POST">
+                        <button class="conversation" name="user" type="submit" value="${match.getId()}" onclick="openChatBox()">
+                            <img src="avatar.png" alt="Person 2" />
+                            <div class="title-text">
+                                <c:out value="${match.getName()}"/>
+                            </div>
+                            <div class="created-date">
+                                April 16
+                            </div>
+                            <div class="conversation-message"> 
+                                This is a message 
+                            </div>
+                        </button>
+                    </form>
+                </c:forEach>
+            </div>
+            <!-- Person 1 chatbox --> 
+            <div id="${recipient}" class="tabcontent active">
+                <div id="chat-title" class="chat-title">
+                    <span> ${recipient} </span>
```

Comment:
When I run this locally, I think this is the line that is displaying a user id. It's probably best if we don't leak user ids. I think you meant to display a user's name here